velox/exec/tests/TableScanTest.cpp

```diff
@@ -105,50 +105,78 @@ class TableScanTest : public HiveConnectorTestBase {
     return getTableScanStats(task).runtimeStats["skippedSplits"].sum;
   }
 
-  void testPartitionedTable(const std::string& filePath) {
+  void testPartitionedTableImpl(
+      const std::shared_ptr<HiveConnectorSplit> split,
+      const std::optional<std::string>& partitionValue) {
+    std::unordered_map<std::string, std::optional<std::string>> partitionKeys =
+        {{"ds", partitionValue}};
     auto outputType = ROW({"ds", "c0", "c1"}, {VARCHAR(), BIGINT(), DOUBLE()});
-
     auto tableHandle = makeTableHandle(SubfieldFilters{});
-
     ColumnHandleMap assignments = {
         {"ds", partitionKey("ds")},
         {"c0", regularColumn("c0")},
         {"c1", regularColumn("c1")}};
 
-    std::unordered_map<std::string, std::optional<std::string>> partitionKeys =
-        {{"ds", {"2020-11-01"}}};
-    auto split = std::make_shared<HiveConnectorSplit>(
-        kHiveConnectorId,
-        filePath,
-        facebook::dwio::common::FileFormat::ORC,
-        0,
-        fs::file_size(filePath),
-        partitionKeys);
-
     auto op = PlanBuilder()
                   .tableScan(outputType, tableHandle, assignments)
                   .planNode();
-    assertQuery(op, split, "SELECT '2020-11-01', * FROM tmp");
+
+    std::string partitionValueStr =
+        partitionValue.has_value() ? "'" + *partitionValue + "'" : "null";
+    assertQuery(
+        op, split, fmt::format("SELECT {}, * FROM tmp", partitionValueStr));
 
     outputType = ROW({"c0", "ds", "c1"}, {BIGINT(), VARCHAR(), DOUBLE()});
     op = PlanBuilder()
              .tableScan(outputType, tableHandle, assignments)
              .planNode();
-    assertQuery(op, split, "SELECT c0, '2020-11-01', c1 FROM tmp");
-
+    assertQuery(
+        op,
+        split,
+        fmt::format("SELECT c0, {}, c1 FROM tmp", partitionValueStr));
     outputType = ROW({"c0", "c1", "ds"}, {BIGINT(), DOUBLE(), VARCHAR()});
     op = PlanBuilder()
              .tableScan(outputType, tableHandle, assignments)
              .planNode();
-    assertQuery(op, split, "SELECT c0, c1, '2020-11-01' FROM tmp");
+    assertQuery(
+        op,
+        split,
+        fmt::format("SELECT c0, c1, {} FROM tmp", partitionValueStr));
 
     // select only partition key
     assignments = {{"ds", partitionKey("ds")}};
     outputType = ROW({"ds"}, {VARCHAR()});
     op = PlanBuilder()
              .tableScan(outputType, tableHandle, assignments)
              .planNode();
-    assertQuery(op, split, "SELECT '2020-11-01' FROM tmp");
+    assertQuery(
+        op, split, fmt::format("SELECT {} FROM tmp", partitionValueStr));
+  }
+
+  void testPartitionedTable(const std::string& filePath) {
+    
+    // Partition column is non-null
+    std::unordered_map<std::string, std::optional<std::string>> partitionKeys =
+        {{"ds", {"2020-11-01"}}};
+    auto split = std::make_shared<HiveConnectorSplit>(
```

Comment:
The split creation logic is duplicated. Any reason not to move it inside of testPartitionedTableImpl?

Reply:
We can move it in. The reason is I felt like split is a better abstraction level rather than just partition keys.